sys/kern/vfs_syscalls.c

```diff
@@ -179,7 +181,36 @@ int do_unlink(proc_t *p, char *path) {
 }
 
 int do_mkdir(proc_t *p, char *path, mode_t mode) {
-  return ENOTSUP;
+  int error;
+  vattr_t va;
+  vnode_t *vn, *dvn;
+  componentname_t cn;
+
+  if ((error = vnr_create(path, &dvn, &cn)))
+    return error;
+
+  if (cn.cn_namelen > NAME_MAX) {
```

Comment:
Should not that case be handled by `vfs_nameresolve` or `vfs_nameresolve`?

Reply:
Moved to `vfs_namecreate`.